File: leetCode/fukun/001-100/lc_069.js
// 二分法
const mySqrt = function (x) {
    if (x === 1) return 1;
    let left = 0, right = x;
    while (1) {
        let currN = Math.floor((left + right) / 2);
        if (currN === left) return left;
        currN * currN > x ? (right = currN) : (left = currN);
    }
};
console.log(mySqrt(9));

File: leetCode/XSS/lc_069.java
public class lc_069 {
    public static void main(String[] args) {
//平方根问题，可用二分法以及牛顿迭代法，使用牛顿迭代法应理解推导公式 ***
    }
    static class Solution {
        public int mySqrt(int x) {
            long left = 0;
            long right = x / 2 + 1;
            while (left < right) {
                long mid = (left + right + 1) >>> 1;
                long square = mid * mid;
                if (square > x) {
                    right = mid - 1;
                } else {
                    left = mid;
                }
            }
            return (int) left;
        }
    }

}
